### PR TITLE
Add a new job for Spark on GKE CI testing

### PR DESCRIPTION
**Pull request by foxish**
Adding a new job for testing the Kubernetes scheduler backend in Apache Spark.

cc/ @cjwagner @kimoonkim @liyinan926 @krzyzacy

### CODE REVIEW

**Comment by BenTheElder** [comment]
> AssertionError: prow/config.yaml is not sorted, please run `bazel run //jobs:config_sort`

We have some presubmit requiring things to stay sorted, this command will auto-sort them.

File: prow/config.yaml [hunk 1]
```diff
@@ -24108,6 +24108,26 @@ periodics:
       secret:
         secretName: fejta-bot-token
 
+- interval: 1h
+  agent: kubernetes
+  name: spark-k8s-periodic
+  spec:
+    containers:
+    # TODO(foxish): Replace latest with a specific tag once the images stabilize.
+    - image: kubespark/test-builder:latest
```
Comment (cjwagner) [inline prow/config.yaml]: Do you need an image repository prefix here (`gcr.io/kubespark/test-builder:latest`)?

Comment (foxish) [reply prow/config.yaml]: It should pick up `docker.io/kubespark/...` by default. I don't mind hosting on gcr, but was unclear on what project would own it.

**Comment by foxish** [comment]
Sorted, @BenTheElder thanks!

**Comment by foxish** [comment]
Tests pass - @BenTheElder, @cjwagner PTAL.
Question - can I add an emptydir volume to the spec for the job here? 
The test image will be cloning repos and running tests - might be much faster with an emptydir over the image's storage layer.

**Comment by foxish** [comment]
Also, need to build images within this job - is docker-in-docker still supported?

**Comment by BenTheElder** [comment]
We do have emptyDir and docker-in-docker is supported though we prefer to
avoid it if possible (managing the data root / base image caching is
slightly problematic)

File: jobs/config.json [hunk 2]
```diff
@@ -318,6 +318,25 @@
...
+      "--env-file=jobs/platform/gke.env",
+      "--extract=ci/latest",
+      "--gcp-node-image=gci",
+      "--gcp-zone=us-central1-f",
+      "--provider=gke",
+      "--test=false",
+      "--timeout=180m",
+      "--gke-environment=prod",
+      "--test-name=spark-integration",
+      "--test-cmd='cd spark-integration && ./e2e/e2e-prow.sh'"
```
Comment (krzyzacy) [inline jobs/config.json]: I think the file tree looks like
```
workspace
- spark-integration
  - (stuff inside spark-integration repo)
  - kubernetes
```

Comment (foxish) [reply jobs/config.json]: It wouldn't land me within `workspace/`?

Comment (krzyzacy) [reply jobs/config.json]: I believe `--extract` will cd into kubernetes

**Comment by krzyzacy** [comment]
(and add `--deployment=gke` to config.json)

**Comment by foxish** [comment]
@krzyzacy tested locally using https://github.com/kubernetes/test-infra#automated-testing. Looks like it works as expected. This is ready for another look.

File: jobs/config.json [hunk 3]
```diff
@@ -11200,5 +11200,24 @@
...
+      "--deployment=gke",
+      "--extract=gke",
+      "--gcp-node-image=gci",
+      "--gcp-zone=us-central1-f",
+      "--gke-environment=prod",
+      "--provider=gke",
+      "--test=false",
+      "--test-cmd=../e2e/e2e-prow.sh",
+      "--test-cmd-name=spark-integration",
+      "--timeout=180m",
```
Comment (krzyzacy) [inline jobs/config.json]: (remove comma, it's json)

File: prow/config.yaml [hunk 4]
```diff
@@ -24108,6 +24108,47 @@ periodics:
...
+  spec:
+    containers:
+    - image: gcr.io/k8s-testimages/kubekins-e2e:v20171218-a18c352b4-1.8
+      imagePullPolicy: Always
+      args:
+      - "--clean"
+      - "--job=$(JOB_NAME)"
+      - "--repo=github.com/apache-spark-on-k8s/spark-integration=cloud-testing-scripts"
+      - "--upload=gs://kubernetes-jenkins/logs/"
+      - "--timeout=75"
```
Comment (krzyzacy) [inline prow/config.yaml]: this timeout need to be greater than config.json ( > 180m, probably 200m)

File: prow/config.yaml [hunk 5]
```diff
@@ -24108,6 +24108,47 @@ periodics:
       secret:
         secretName: fejta-bot-token
 
+- interval: 1h
+  agent: kubernetes
+  name: spark-k8s-periodic
+  spec:
+    containers:
+    - image: gcr.io/k8s-testimages/kubekins-e2e:v20171218-a18c352b4-1.8
+      imagePullPolicy: Always
```
Comment (krzyzacy) [inline prow/config.yaml]: don't need `imagePullPolicy` here

**Comment by krzyzacy** [comment]
looks fine otherwise, please squash

File: prow/config.yaml (hunk 4 above)
Comment (foxish) [reply prow/config.yaml]: Going to do 120m and 150m. That should work right? I don't think it should ever take longer than 2h.

Comment (krzyzacy) [reply prow/config.yaml]: you know better how long it would take :-)

**Pull request merged**